### ISSUE TITLE
Potential Data Leakage Risk for API Models on Private Evaluation Sets

### ISSUE THREAD

**Issue by linyubupa** [huggingface/open_asr_leaderboard#170]
Hi OpenASR team,

I have a concern regarding the private evaluation datasets.

For open-weight models, the training data is fixed and publicly released. However, API-based ASR services are continuously updated and may potentially collect evaluation traffic. Over time, repeated submissions of the same private benchmark samples could lead to indirect exposure of the private dataset, allowing providers to optimize for it intentionally or unintentionally.

One observation that motivated this question is the large improvement of Microsoft Azure Speech between the 05-2026 and 06-2026 releases on the private benchmark. Its average private WER improved from 10.10 to 8.08, with conversational WER improving from 18.18 to 14.63 in a single monthly release. Such improvements may be entirely legitimate, but they also highlight the difficulty of ensuring that a fixed private benchmark remains truly unseen for continuously updated API services.

If private benchmark samples are repeatedly evaluated against the same APIs over long periods, there is a risk that benchmark traffic could eventually influence future model updates, intentionally or unintentionally. In that case, the private benchmark would gradually lose its value as an unseen test set, and rankings might no longer reflect true generalization performance.

Has the leaderboard considered this issue? Are there any safeguards (e.g., dataset rotation, refreshes of the private evaluation sets, or agreements that evaluation traffic is not used for training) to prevent private data leakage for API models?

I am not accusing any specific provider of misconduct. Rather, I believe this is an important benchmarking challenge for all continuously updated closed-source ASR services.

Thanks.

<img width="3148" height="1224" alt="Image" src="https://github.com/user-attachments/assets/4f58c399-f65a-4271-a625-b29dd42c7b8a" />

<img width="1547" height="473" alt="Image" src="https://github.com/user-attachments/assets/586390be-bb8f-404f-96dc-04dad5ea618f" />

**Comment by ebezzam** [comment]
Hi @linyubupa thank you for the comment and the valid concern!

Firstly, on the apparant "improvement" of `microsoft/azure-speech-06-2026`:
- The same endpoint was used in both cases, as pinned [here](https://github.com/huggingface/open_asr_leaderboard/blob/4e4880b9fb203d60830ed2920c521e067026f269/api/providers/microsoft_azure_provider.py#L14). The date change is to reflect when we evaluated.
- The "improvement" rather came from our error in evaluating the model. Our private evaluation was missing [this prompt](https://github.com/huggingface/open_asr_leaderboard/blob/main/api/run_api.sh#L64-L67) for `05-2026` which was needed by the private set.
- With #142, we also made normalizer improvements which improved all models from (13-06-2026) to (24-06-2026).

Second, you bring up some very valid concerns! Answering below:

> API-based ASR services are continuously updated and may potentially collect evaluation traffic. Over time, repeated submissions of the same private benchmark samples could lead to indirect exposure of the private dataset, allowing providers to optimize for it intentionally or unintentionally.

This is something very hard for us to control. We don't plan to perform repeated evaluations of the same set. Although for the "improvement" you observed we had to repeat evaluations on **all models**, because:
1. The old evaluation scripts saved normalized predictions. And since we updated the normalizer with #142, we had to recompute all models outputs. But now with additional changes in #142, unnormalized outputs are saved (for example [here](https://github.com/huggingface/open_asr_leaderboard/blob/4e4880b9fb203d60830ed2920c521e067026f269/api/run_eval.py#L186-L187)) in an HF bucket (set [here](https://github.com/huggingface/open_asr_leaderboard/blob/4e4880b9fb203d60830ed2920c521e067026f269/api/submit_jobs.sh#L17)). So we don't need to pass the same files into the API models in case there are future changes to the normalizer.  
2. In particular, for the Microsoft azure model as it was incorrectly evaluated, as mentioned earlier.

> If private benchmark samples are repeatedly evaluated against the same APIs over long periods, there is a risk that benchmark traffic could eventually influence future model updates, intentionally or unintentionally. In that case, the private benchmark would gradually lose its value as an unseen test set, and rankings might no longer reflect true generalization performance.

As mentioned above, we don't plan for repeated evaluations. For the current private sets, API models had two passes of the private sets because of the normalizer update and our old workflow of saving normalized outputs.

> Are there any safeguards (e.g., dataset rotation, refreshes of the private evaluation sets, or agreements that evaluation traffic is not used for training) to prevent private data leakage for API models?

There are planned additions of the private evaluations set, such that the weight of the current ones will diminish. And now that unnormalized predictions are saved, API models should only see the new private data once.

---

We can leave this issue open, in case others have the same question or ideas to reduce the risk of API models from having an advantage by continuous training. Thanks again for raising this point!

**Comment by BrickDesignerNL** [comment]
Hi @ebezzam , as you requested on LinkedIn:

> [Wouter van der Harg](https://www.linkedin.com/in/vanderharg) and if you (or anyone else) has ideas or thoughts on private data leakage, there's a GitHub issue started here on this topic! [https://github.com/huggingface/open_asr_leaderboard/issues/170](https://github.com/huggingface/open_asr_leaderboard/issues/170)

The following might be useful:
https://deepmind.google/blog/piloting-the-worlds-first-double-blind-ai-evaluations/

**Comment by MarktHart** [comment]
I don't think this is a big issue. For an API provider, it's literally stealing customer data. That's a big deal. I see no reason why you would ever do that, let alone just to fake a better model. If stealing customer data was an option, you wouldn't need to fake a better model, you could just take the top spot by throwing training data at the problem.

For the same reason I assume Huggingface isn't stealing our (closed) model, despite those weights being hosted on Huggingface.

On the Deepmind article: don't try to math your way out of simply signing a contract, this is already a solved problem.

**Comment by BrickDesignerNL** [comment]
@MarktHart not sure if you could call it stealing customer data, it depends on the terms of use. For example if terms state that the API provider is allowed to improve it's services based on your usage and you agreed.

I did get a certain cloud hosted AI model to inform me about a model hosted on hugging face that isn't public (yet). And no it was not the one that hacked 🤗 recently. And no the information is not on internet or other public source. I have validated the information with the owner of the private model. So it might be that private space on hugging face is less closed then intended.

Do you sign contracts with all parties before running a benchmark? Wouldn't that defeat the purpose; add the risk of getting extra compute power and a less quantized version than regularly?

**Comment by MarktHart** [comment]
@BrickDesignerNL 

> Do you sign contracts with all parties before running a benchmark? 

Benchmarking parties sign up like any other customer, so we do have some standard TOS. Customers (or parties running benchmarks) wanting to run our model on-premise on their own GPUs do need a contract.

> Wouldn't that defeat the purpose; add the risk of getting extra compute power and a less quantized version than regularly?

This would also be the problem with the enclave thing of Deepmind?
